dbms/src/Interpreters/Join.cpp

```diff
@@ -1911,9 +1987,16 @@ void Join::checkTypesOfKeys(const Block & block_left, const Block & block_right)
     }
 }
 
-void Join::joinBlock(Block & block) const
+Block Join::joinBlock(size_t stream_index) const
+{
+    Block block = probe_process_infos[stream_index]->block;
+    joinBlock(block, stream_index);
+    return block;
+}
+
+void Join::joinBlock(Block & block, size_t stream_index) const
 {
-    //    std::cerr << "joinBlock: " << block.dumpStructure() << "\n";
+    RUNTIME_CHECK(probe_initialized);
```

Comment:
it should moved to L2011, under `std::shared_lock lock(rwlock);`.

Reply:
done.